File: rb/features.rb
# Listenfree SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ListenfreeFeatures
  def self.make_feature(name)
    case name
    when "base"
      ListenfreeBaseFeature.new
    when "ratelimit"
      ListenfreeRatelimitFeature.new
    when "retry"
      ListenfreeRetryFeature.new
    when "test"
      ListenfreeTestFeature.new
    when "timeout"
      ListenfreeTimeoutFeature.new
    else
      ListenfreeBaseFeature.new
    end
  end
end

File: py/features.py
# Listenfree SDK feature factory

from feature.base_feature import ListenfreeBaseFeature
from feature.test_feature import ListenfreeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ListenfreeBaseFeature(),
        "test": lambda: ListenfreeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
